test/check-application

```diff
@@ -349,21 +349,38 @@ class TestApplication(testlib.MachineCase):
         b.set_input_text('.publish-port-form input:last-child', '6000')
         b.click('.publish-port-form .fa-plus')
         b.wait_present('.publish-port-form + .publish-port-form')
-        b.set_input_text('.publish-port-form:last-of-type input:first-child', '5001')
-        b.set_input_text('.publish-port-form:last-of-type input:last-child', '6001')
-        b.click('.publish-port-form:last-of-type button.dropdown-toggle')
-        b.wait_visible('.publish-port-form:last-of-type .dropdown-menu a:contains("UDP")')
-        b.click('.publish-port-form:last-of-type .dropdown-menu a:contains("UDP")')
-        b.wait_visible('.publish-port-form:last-of-type button.dropdown-toggle:contains("UDP")')
-        b.click('.publish-port-form:last-of-type .fa-plus')
+        b.set_input_text('.publish-port-form + .publish-port-form input:first-child', '5001')
+        b.set_input_text('.publish-port-form + .publish-port-form input:last-child', '6001')
```

Comment:
This duplication is a bit ugly. This is because each port *entry* its own form, right?

Could this use a more specific selector instead, such as `.publish-port-form[containerPort='']` to select the first empty one, or in other cases with the expected port to select a filled one?

Reply:
the outer div doesn't really have any  of the attributes of the input fields. An alternative is `nth-child` from the parent of this div.